Chess/include/Board.h

```diff
@@ -0,0 +1,38 @@
+#pragma once
+#include"Piece.h"
+#include<vector>
+#include "Rook.h"
+#include "Bishop.h"
+#include "Knight.h"
+#include "Queen.h"
+#include"King.h"
+#include"Pawn.h"
+#include"Player.h"
+
+class Piece;
+
+/**
+* The Board class represents a chessboard and manages the pieces on it.
+*/
+
+class Board {
+	Player whos_turn = WHITE_PLAYER;
+	Piece* pieces[8][8];
+
+public:
+	Board();
```

Comment:
i don't see a destructor, how do you free the memory ??

Reply:
I had a problem with the destructor so I skipped on it . now I have fixed it